src/components/DateTimePicker/helpers/extractTime.js

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable no-console */
+
+export default function extractTime(date) {
+    if (date) {
+        try {
+            const options = {
+                hour12: true,
+                hour: '2-digit',
+                minute: '2-digit',
+            };
+            const value = typeof date === 'string' ? new Date(date) : date;
+            return new Intl.DateTimeFormat('en-US', options).format(value);
+        } catch (error) {
+            console.error(error);
+            return '';
+        }
+    }
+    console.error('Invalid date value passed to DateTimePicker');
+    return '';
```

Comment:
here I think it is better to extract the time manually thus it is more deterministic since implementations of Intl.DateTimeFormat can differ in browsers and then we ensure the right value is passed to TimeSelect
use date.getHours() and date.getMinutes()
keep in mind that that getHours return only one digit when below 10 so when it returns 2 we should use 02